### ISSUE TITLE
getting exception on pin clicked event

### ISSUE THREAD

**Issue by ShariatPanah** [Mapsui/Mapsui#682]
so i use this code to configure the map in Xamarin.Forms:
```
map = new Map();
map.Layers.Add(OpenStreetMap.CreateTileLayer());

var view = new MapView();
view.HorizontalOptions = LayoutOptions.FillAndExpand;
view.VerticalOptions = LayoutOptions.FillAndExpand;
view.IsMyLocationButtonVisible = true;
view.IsNorthingButtonVisible = true;
view.MyLocationEnabled = false;
view.MyLocationFollow = false;
view.Map = map;
view.IsEnabled = true;

view.PinClicked += (sender, args) =>
{
	DisplayAlert("", $"{args.Point.Latitude}\n{args.Point.Longitude}", "OK");
};

var pin = new Pin(view)
{
	Type = PinType.Pin,
	Label = "Some Pin",
	Position = new Position(36.154498, 45.484425),
	Address = "Address..."
};

view.Pins.Add(pin);

stackLayout.Children.Add(view);
Content = stackLayout;
```

now first of all, when i run the application the map and the pin are shown but when i click on the pin i get this exception at the line of DisplayAlert method:
"Exception has been thrown by the target of an invocation."

the inner message of this exception is saying:
"Can't create handler inside thread that has not called Looper.prepare()"

the most interesting part is when i run the application in emulator the PinClicked and the MapLongClicked events will be fired nicely but in real device they wont, like if they're not implemented at all.

Secondly, a mark of like it would be my location shown at the center of the map, but obviously I'm not at that point and i don't know how to make it disappear:
![sdfsdf](https://user-images.githubusercontent.com/32509267/62234191-6a0fa580-b3df-11e9-958f-dca0efc98ae5.png)

and the third problem i have, the NavigateTo method is not working. i use this code:
```
var centerOfSardasht = new Mapsui.Geometries.Point(36.1584, 45.4760);
var sardashtPoint = SphericalMercator.ToLonLat(36.1584, 45.4760);
view.Navigator.NavigateTo(sardashtPoint, 1000);
```
but it wont do anything, it wont navigate to the point I specified, not with the centerOfSardasht and neither with sardashtPoint variable.
please let me know if I'm missing something, and sorry for telling this but your library is really complicated and hard to work with.

thank you, any advice would be appreciated

**Comment by ShariatPanah** [comment]
hey, is there no one to help?
i managed to fix the exception by using Device.BeginInvokeOnMainThread, and the exception is gone, and the PinClicked event is now get fired on a real device too, but the MapClicked and MapLongClicked events don't get fired on real device, i'm testing on Huawei mate 10 lite ANDROID 8

about the third problem, NavigateTo method is actually does the navigating but not to the point i specified, it always moves to the center of the map where the MyLocation Icon is shown, no matter what coordinates i provide, it always get to the center

if you want me to provide a sample, just let me know.

**Comment by pauldendulk** [comment]
@ShariatPanah Thanks for reporting this. Could you create a fork of the mapsui repo and create a sample of your problem based on one of the mapsui samples?

**Comment by pauldendulk** [comment]
I assume this is related to the callout issue which should be fixed in 2.0.0-beta.34.
